internal/logic/logic.go

```diff
@@ -1,34 +1,201 @@
 package logic
 
 import (
+	"fmt"
+	"github.com/PereRohit/util/log"
+	respModel "github.com/PereRohit/util/model"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
+	"github.com/vatsal278/UserManagementService/internal/codes"
+	"github.com/vatsal278/UserManagementService/internal/config"
+	"github.com/vatsal278/UserManagementService/internal/model"
+	jwtSvc "github.com/vatsal278/UserManagementService/internal/repo/authentication"
+	"github.com/vatsal278/UserManagementService/internal/repo/crypto"
 	"github.com/vatsal278/UserManagementService/internal/repo/datasource"
+	"net/http"
+	"time"
 )
 
-//go:generate mockgen --build_flags=--mod=mod --destination=./../../pkg/mock/mock_logic.go --package=mock github.com/vatsal278/UserManagementService/internal/logic UserManagementServiceLogicIer
+//go:generate mockgen --build_flags=--mod=mod --destination=./../../pkg/mock/mock_logic.go --package=mock github.com/vatsal278/UserManagementService/internal/logic UserMgmtSvcLogicIer
 
 type UserMgmtSvcLogicIer interface {
 	HealthCheck() bool
-	//Signup(model.SignUpCredentials) *respModel.Response
-	//Login(http.ResponseWriter, model.LoginCredentials) *respModel.Response
+	Signup(model.SignUpCredentials) *respModel.Response
+	Login(http.ResponseWriter, model.LoginCredentials) *respModel.Response
 	//UserData(any) *respModel.Response
 	//Activate(id any) *respModel.Response
 }
 
 type userMgmtSvcLogic struct {
-	DsSvc datasource.DataSourceI
-	//loginService jwtSvc.LoginService
-	//jwtService   jwtSvc.JWTService
+	DsSvc      datasource.DataSourceI
+	jwtService jwtSvc.JWTService
+	msgQueue   config.MsgQueue
+	cookie     config.CookieStruct
 }
 
-func NewUserMgmtSvcLogic(ds datasource.DataSourceI) UserMgmtSvcLogicIer {
+func NewUserMgmtSvcLogic(ds datasource.DataSourceI, jwtService jwtSvc.JWTService, msgQueue config.MsgQueue, cookie config.CookieStruct) UserMgmtSvcLogicIer {
 	return &userMgmtSvcLogic{
-		DsSvc: ds,
-		//loginService: loginSvc,
-		//jwtService:   jwtService,
+		DsSvc:      ds,
+		jwtService: jwtService,
+		msgQueue:   msgQueue,
+		cookie:     cookie,
 	}
 }
 
 func (l userMgmtSvcLogic) HealthCheck() bool {
 	// check all internal services are working fine
 	return l.DsSvc.HealthCheck()
 }
+
+func (l userMgmtSvcLogic) Signup(credential model.SignUpCredentials) *respModel.Response {
+	result, err := l.DsSvc.Get(map[string]interface{}{"email": credential.Email})
+	if err != nil {
+		log.Error(err.Error())
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrCreatingAccount),
+			Data:    nil,
+		}
+	}
+	if len(result) != 0 {
+		log.Error(codes.GetErr(codes.ErrEmailExists))
+		return &respModel.Response{
+			Status:  http.StatusBadRequest,
+			Message: codes.GetErr(codes.ErrEmailExists),
+			Data:    nil,
+		}
+	}
+	newUser := model.User{
+		Id:           uuid.New().String(),
+		Email:        credential.Email,
+		Name:         credential.Name,
+		Company:      "perennial",
```

Comment:
Keep the company name as empty. This will be added later when the user edits their details.

Reply:
done